clients/src/main/java/org/apache/kafka/clients/consumer/internals/WakeupTrigger.java

```diff
@@ -43,8 +43,12 @@ public void wakeup() {
                 return new WakeupFuture();
             } else if (task instanceof ActiveFuture) {
                 ActiveFuture active = (ActiveFuture) task;
-                active.future().completeExceptionally(new WakeupException());
-                return null;
+                boolean wasTriggered = active.future().completeExceptionally(new WakeupException());
```

Comment:
Could we add a test in `WakeupTriggerTest` for this?

Reply:
Will do.

Reply:
I added some unit tests.